### PR TITLE
Update SchedulerMinHeap

### PR DESCRIPTION
**Pull request by JX-Zhuang**
## Summary
Update SchedulerMinHeap.Rewrite siftUp and siftDown.

## How did you test this change?

yarn test 
yarn test scheduler

### CODE REVIEW

**Comment by salazarm** [comment]
I don't see how this is more optimal?

**Comment by JX-Zhuang** [comment]
> I don't see how this is more optimal? Care to explain? Looks like it's just moving the logic around making it potentially less readable

Use zero-fill left shift instead of multiplication.Move the common logic making it shorter and single responsibility.Less `if...else`.
``` javascript
function siftUp(heap, node, i) {
  let index = i;
  let parentIndex = (index - 1) >>> 1;
  // index > 0 and parent is larger. Swap.
  while (index > 0 && compare(heap[parentIndex], node) > 0) {
    swap(heap, parentIndex, index);
    index = parentIndex;
    parentIndex = (index - 1) >>> 1;
  }
}
function siftDown(heap, node, i) {
  let index = i;
  const length = heap.length;
  const halfLength = length >>> 1;
  while (index < halfLength) {
    // Use zero-fill left shift
    const leftIndex = (index << 1) + 1;
    const rightIndex = leftIndex + 1;
    // First,find least index.Assume left child is least
    let nextIndex = leftIndex;
    if (rightIndex < length && compare(heap[rightIndex], heap[leftIndex]) < 0) {
      nextIndex = rightIndex;
    }
    // Then,if least>=node, exit.
    if (compare(heap[nextIndex], node) >= 0) return;
    // Last,swap node and least
    swap(heap, nextIndex, index);
    index = nextIndex;
  }
}
```
Refer `siftDownComparable` in [`PriorityQueue.java`].(http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/PriorityQueue.java)

**Comment by JX-Zhuang** [comment]
> I don't see how this is more optimal? Care to explain? Looks like it's just moving the logic around making it potentially less readable

I add some annotations making it more readable.

**Comment by JX-Zhuang** [comment]
@acdlite please review

File: packages/scheduler/src/SchedulerMinHeap.js [hunk 1]
```diff
@@ -90,3 +72,8 @@ function compare(a, b) {
   const diff = a.sortIndex - b.sortIndex;
   return diff !== 0 ? diff : a.id - b.id;
 }
+function swap(heap: Heap, indexA, indexB) {
+  const temp = heap[indexA];
+  heap[indexA] = heap[indexB];
+  heap[indexB] = temp;
```
Comment (SukkaW) [inline packages/scheduler/src/SchedulerMinHeap.js:78]:
```suggestion
  [heap[indexB], heap[indexA]] = [heap[indexA], heap[indexB]]
```

Even smaller~

**Pull request closed without merging**